### DISCUSSION TITLE
How do I get RSSI of a device I am already connected to.

### DISCUSSION THREAD

**Discussion by EdwinFairchild** [hbldh/bleak#879] [Q&A]
I am connecting to a device like so:
`async with BleakClient(self.ble_address, disconnected_callback= self.handle_disconnect) as client:`

Now I want be able to get the RSSI from my device periodically, perhaps use it to get a rough estimate of distance  
from computer , just as an example... But I am unsure the proper method to get the RSSI once already connected. I have seen
code around here but they are using the scanner object, but once we have scanned and connected how do you continue to get RSSI ?

**Comment by dlech** [comment]
This is a case of a feature that is not currently available on all OSes. 

- Windows is missing: https://stackoverflow.com/a/65711336/1976323. 
- macOS is available: https://developer.apple.com/documentation/corebluetooth/cbperipheral/1519111-readrssi?language=objc
- Linux/BlueZ: RSSI D-Bus property doesn't seem to be updated after connection.
- Android is available: https://developer.android.com/reference/android/bluetooth/BluetoothGatt#readRemoteRssi()

If you control the device firmware, you could add a characteristic for reading the RSSI from the remote device.

**Reply by EdwinFairchild** [reply]
oh yes that is true, I can just put it in a characteristic, simple but effective! thanks!

**Reply by mr-manuel** [reply]
I'm using Linux and would need the same one. How did you do that exactly?

**Reply by EdwinFairchild** [reply]
Well in my case I have control over the firmware running on the bluetooth device so I added a characteristic that I can use to request the RSSI, nothing special about the characteristic itself its a simple data characteristic .

**Reply by mr-manuel** [reply]
Thanks. I found also out, that the RSSI on some devices is only displayed, if I execute `bluetoothctl scan on` in background, else `bluetoothctl info MACADDRESS` is not showing the RSSI.
